src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -472,6 +472,39 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
         matrixType,
         MatrixValue(matrixType, globals, colValues, rvd)))
 
+  private type Axis = Int
+  private val globalAxis: Axis = 0
+  private val rowAxis: Axis = 1
+  private val colAxis: Axis = 2
+  private val entryAxis: Axis = 3
```

Comment:
generally I prefer `private[this] ` whenever possible (there's very minor performance improvements for field accesses [b/c/ there's no indirection])